### PR TITLE
Special control charts

### PR DESCRIPTION
**Pull request by JTPetter**
- Complete rewrite of time weighted charts (cusum & ewma chart), verified and with more functionality
- Complete rewrite of rare event charts (g & t chart), verified and with more functionality
- Help files for time weighted and rare event charts
- Fixes https://github.com/jasp-stats/jasp-issues/issues/2895

### CODE REVIEW

**Comment by JTPetter** [comment]
@vandenman Excuses for the big PR, but I could not find a good way of splitting this (without submitting something that is not functional). But most lines and files are related to the plots in the unit tests anyways, so not quite as big as it might appear.

File: R/commonQualityControl.R [hunk 1]
```diff
@@ -787,13 +912,20 @@ KnownControlStats.RS <- function(N, sigma = 3) {
...
-                    "I"    = "Individual value",
-                    "MR"   = "Moving range",
-                    "MMR"  = "Moving range of subgroup mean",
-                    "s"    = "Sample std. dev.")
+  if (plotType == "g" || plotType == "t") {
+    unitString <- paste0(toupper(substr(gAndtUnit, 1, 1)), substr(gAndtUnit, 2, nchar(gAndtUnit)))
+    yTitle <- gettextf("%1$s between events", unitString)
+  } else {
+    yTitle <- switch (plotType,
+                      "xBar"  = "Sample average",
```
Comment (vandenman) [inline R/commonQualityControl.R:920]: do these need gettext?

File: inst/qml/rareEventCharts.qml [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+					{ label: qsTr("HM"),							value: "HM"},
+					{ label: qsTr("HMS"),							value: "HMS"},
+					{ label: qsTr("Ip"),							value: "Ip"},
+					{ label: qsTr("IMp"),							value: "IMp"}				
+				]
+				indexDefaultValue: 1
+			}
+		}
+
+	}
```
Comment (vandenman) [inline inst/qml/rareEventCharts.qml:138]: @JorisGoosen don't you suddenly get an urge to create a date/ datetime column when you see this? :stuck_out_tongue:

File: R/commonQualityControl.R [hunk 3]
```diff
@@ -326,9 +326,9 @@ KnownControlStats.RS <- function(N, sigma = 3) {
   Data.d2 <- data.frame(
```
[the remark this answers is not in the document]
Comment (JTPetter) [reply R/commonQualityControl.R:326]: added a reference

File: R/commonQualityControl.R (hunk 1 above)
Comment (JTPetter) [reply R/commonQualityControl.R:920]: yes

**Comment by JTPetter** [comment]
@vandenman Thanks! I have addressed your feedback.

**Pull request merged**